api/src/schema/resolvers.js

```diff
@@ -31,7 +41,40 @@ const resolvers = {
         )
       }
 
-      let query = createQuery(forwardSortationArea, filter)
+      let query = {
+        $and: [
+          {
+            forwardSortationArea,
+          },
+        ],
+      }
+
+      if (filter) {
+        for (let key in filter) {
+          // make sure the key exists in the comparators' keys
+          if (Object.keys(comparators).includes(key)) {
+            // if the value is a number
+            // - optionally starting with a minus sign
+            // - containing zero or one decimal places
+            // convert it to a float instead of a string
+            if (filter[key].match(/^-?\d+\.?\d+$/)) {
+              filter[key] = parseFloat(filter[key])
+            }
+
+            // We are evaling our own code here, not user input.
+            // Filter.field is our code, stringified and stored in an enum.
+            // The user choses one of the enum values and we convert the string
+            // back to a function which accepts a matcher and generates the
+            // query we need to find that field.
+            // eslint-disable-next-line no-eval
+            let queryGenerator = eval(filter.field)
```

Comment:
Ignore me if this is a naive comment, but is there a way to import './enums' as a module with and then refer to methods in it using a string?

Ie, `enums['dwellingHouseId']`, or something?

Because at the minute, _we_ know that `filter.field` is one of the enums imported at the top of the file, but it's not immediately obvious to the casual reader.

Reply:
`import * as enums from './enums'` would give exactly that: `enums['dwellingHouseId']`.
The enums are the only values possible in there according to the type system. I think what you are getting at here is that it could/should be a little more obvious what's going on. I have some ideas for that I am about to start on.

Reply:
Yeah, that's what I meant. 👍 if you're working on it.